src/bin/main.rs

```diff
@@ -0,0 +1,133 @@
+use clap::{App, Arg, SubCommand};
+
+mod config;
+use config::Config;
+use fitbit::User;
+
+fn main() -> Result<(), Error> {
+    let default_dir = Path::new(&env::var("HOME")?).join(".config/fitbit-grabber");
+    let default_config = default_dir.clone().join("conf.toml");
+    let date_arg = Arg::with_name("date")
+        .long("date")
+        .required(true)
+        .takes_value(true)
+        .help("date to fetch data for");
+
+    let matches = App::new("Fitbit Grabber")
+        .arg(
+            Arg::with_name("config")
+                .help("path to config file")
+                .short("c")
+                .long("config")
+                .default_value(default_config.to_str().unwrap()),
+        )
+        .subcommand(
+            SubCommand::with_name("heart")
+                .about("fetch heart data")
+                .arg(date_arg.clone()),
+        )
+        .subcommand(
+            SubCommand::with_name("step")
+                .about("fetch step data")
+                .arg(date_arg.clone()),
+        )
+        .subcommand(
+            SubCommand::with_name("weight")
+                .about("fetch body weight data")
+                .arg(date_arg.clone()),
+        )
+        .subcommand(SubCommand::with_name("token").about("request an access token"))
+        .subcommand(SubCommand::with_name("refresh-token").about("refresh token"))
+        .subcommand(SubCommand::with_name("user").about("get user profile"))
+        .get_matches();
+
+    let conf = Config::load(matches.value_of("config"))?;
+    let config::FitbitConfig {
+        client_id,
+        client_secret,
+    } = conf.fitbit.unwrap();
+
+    let auth = fitbit::FitbitAuth::new(&client_id.unwrap(), &client_secret.unwrap());
+
+    if let Some(_) = matches.subcommand_matches("token") {
+        let token = auth.get_token()?;
+        save_token(".token", token)?;
+    }
+
+    if let Some(_) = matches.subcommand_matches("refresh-token") {
+        let token = load_token(".token")?;
+        let exchanged = auth.exchange_refresh_token(token)?;
+        save_token(".token", exchanged)?;
+    }
+
+    let token = load_token(".token").unwrap();
+    let f = fitbit::FitbitClient::new(token)?;
+
+    if let Some(matches) = matches.subcommand_matches("heart") {
+        let raw_date = matches
+            .value_of("date")
+            .ok_or(format_err!("please give a starting date"))?;
+        let date = NaiveDate::parse_from_str(&raw_date, "%Y-%m-%d")?;
+        let heart_rate_data = f.heart(date)?;
+        println!("{}", heart_rate_data);
+    }
+
+    if let Some(matches) = matches.subcommand_matches("step") {
+        let raw_date = matches
+            .value_of("date")
+            .ok_or(format_err!("please give a starting date"))?;
+        let date = NaiveDate::parse_from_str(&raw_date, "%Y-%m-%d")
+            .map_err(|e| format_err!("could not parse date {}", e))?;
+        let step_data = f.step(date)?;
+        println!("{}", step_data);
```

Comment:
This method, `step`, still currently returns a string. If we model it to return library structs, we will need to re-serialize json here to preserve the same behavior as before. Serde might print slightly different after reserialization.